keps/sig-instrumentation/383-new-event-api-ga-graduation/README.md

```diff
@@ -439,6 +449,198 @@ List Events from the NamespaceSystem with field selector `reportingController =
+
+      No.
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+
+  No default behavior will be changed.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Also set `rollback-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g. if this is runtime
+  feature, can it break the existing applications?).
+
+  Yes. If the new Event API is disabled, it will fallback to the original one.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  New types of Events will be generated instead of the old one.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling and disabling feature
+  gates. However, unit tests in each component dealing with managing data created
+  with and without the feature are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+  Manual tests will be performed to ensure things work when either enabling or disabling the new Event API.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+  A rollout could fail if some components restart in the middle of the rollout. Then those components will continue using the old Event API.
+
+* **What specific metrics should inform a rollback?**
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and do that now.
+
+  Not yet. It could be done by enabling / disabling new Event API.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features,
+  APIs, fields of API types, flags, etc.?**
+  Even if applying deprecation policies, they may still surprise some users.
+
+  State field of EventSeries will be removed from corev1.Event API.
+
+### Monitoring requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metrics. Operations against Kubernetes API (e.g.
+  checking if there are objects with field X set) may be last resort. Avoid
+  logs or events for this purpose.
+
+  There's no way to tell that through an operator. If a component has been migrated to use the new Event API and the API is enabled, it should use that by default. Currently only scheduler has been migrated.
```

Comment:
This isn't a "feature for workloads" per-se.

I think again we need to separate two different things:
- the API
- the client libraries.

The former - workloads may in theory use, the latter, is not a workload feature.

The former can be determined by looking into apiserver_requests_total metric.

Reply:
"operator" here refers to a human, the answer sounds like it interpreted it as referring to an automated operator/controller.

Reply:
> "operator" here refers to a human, the answer sounds like it interpreted it as referring to an automated operator/controller.

Right, I thought "operator" refers to some watchers running in the cluster.